VENDORING.md

```diff
@@ -0,0 +1,235 @@
+# Performing vendoring for the Kata Containers project
+
+* [Vendoring tool](#vendoring-tool)
+    * [How to install and update dep](#how-to-install-and-update-dep)
+    * [Overview of the main commands](#overview-of-the-main-commands)
+* [Vendoring use cases](#vendoring-use-cases)
+    * [Initialize vendoring of the repository](#initialize-vendoring-of-the-repository)
+    * [Vendor a newly imported package](#vendor-a-newly-imported-package)
+    * [Update a vendored package](#update-a-vendored-package)
+    * [Remove a vendored package](#remove-a-vendored-package)
+    * [Test a pending pull request](#test-a-pending-pull-request)
+* [Miscellaneaous](#miscellaneaous)
+
+## Vendoring tool
+
+[`dep`](https://github.com/golang/dep) is the vendoring tool chosen for this
+project. The choice has been made to use this tool because it is fairly simple
+and very powerful for some corner cases. Furthermore, this tool is expected to
+become the official vendoring tool for Golang.
+
+Ensure you are using the latest version of the `dep` tool. See 
+[How to install and update dep](#how-to-install-and-update-dep) for details.
+
+### How to install and update dep
+
+Installing or updating this tool is simple, use the following command:
+```bash
+$ go get -u github.com/golang/dep/cmd/dep
+```
+
+Make sure your `PATH` contains `$GOPATH/bin`:
+```bash
+$ export PATH=$GOPATH/bin:$PATH
+```
+
+### Overview of the main commands
+
+- `dep init`: Initialize a new project with manifest and lock files.
+- `dep ensure`: Ensure a dependency is safely vendored in the project.
+- `dep prune`: Prune the vendor tree of unused packages.
+
+## Vendoring use cases
+
+Here will be listed the different vendoring use cases a developer could
+encounter when developing for Kata Containers.
+
+### Initialize vendoring of the repository
+
+If no prior vendoring has been performed on the repository, you need to
+initialize it with the following command:
+```bash
+$ dep init
+```
+
+This should create a new directory named `vendor` at the root of your
+repository, and two new files to manage the vendoring: `Gopkg.toml` and
+`Gopkg.lock`.
+
+`Gopkg.lock` is automatically re-generated by `dep` every time a new
+`dep ensure` command is issued. This file should not be modified by the
+developer.
+`Gopkg.toml` is automatically generated by `dep init`, and can be modified
+in order to constraint any package to follow a specific revision.
```

Comment:
maybe this should be "to add a constraint for a package.."

Reply:
done